server/src/main/scala/org/http4s/server/middleware/Timeout.scala

```diff
@@ -33,15 +37,18 @@ object Timeout {
 
   /** Transform the service to return a RequestTimeOut [[Status]] after the supplied Duration
     * @param timeout Duration to wait before returning the RequestTimeOut
+    * @param customScheduler a custom scheduler to use for timing out the request.
+    *   If `None` is provided, then a default daemon thread scheduler will be used.
     * @param service [[HttpService]] to transform
     */
-  def apply(timeout: Duration, response: Task[Response] = DefaultTimeoutResponse)(service: HttpService): HttpService = {
-    if (timeout.isFinite()) apply(timeoutResp(timeout, response))(service)
+  def apply(timeout: Duration, response: Task[Response] = DefaultTimeoutResponse, customScheduler: Option[ScheduledExecutorService] = None)(service: HttpService): HttpService = {
```

Comment:
I guess if we want to change it, we might as well while we are breaking the compatibility of the method. It does seem like if you are calling this you probably always want a `FiniteDuration`, but it's also handing the infinite case gracefully, so it didn't seem too bad to me in its current form. Shall I go ahead and change it then?